chat/chat-tool/tool/src/java/org/sakaiproject/chat2/tool/DecoratedChatChannel.java

```diff
@@ -179,33 +185,68 @@ public int getNumberChannelMessages() {
       return chatTool.countChannelMessages(chatChannel);
    }
 
-   public Date getStartDate() {
-       return startDate;
-   }
-
-   public void setStartDate(Date startDate) {
-       if (startDate != null) {
-           // fix up the date to shift to be beginning or end of the day (drop any time component)
-           startDate = DateUtils.truncate(startDate, Calendar.DATE);
-       }
-       this.startDate = startDate;
-   }
-
-   public Date getEndDate() {
-       if (endDate != null) {
-           // fix up the date to drop any time component
-           endDate = DateUtils.truncate(endDate, Calendar.DATE);
-       }
-       return endDate;
-   }
-
-   public void setEndDate(Date endDate) {
-       if (endDate != null) {
-           // fix up the date to shift to be beginning or end of the day (drop any time component)
-           endDate = DateUtils.truncate(endDate, Calendar.DATE);
-           endDate = DateUtils.addSeconds(endDate, 86398); // just short of a full day in seconds
-       }
-       this.endDate = endDate;
-   }
-
+	public Date getStartDate() {
+		return startDate;
+	}
+
+	public void setStartDate(Date startDate) {
+		this.startDate = startDate;		
+	}
+
+	public Date getEndDate() {
+		return endDate;
+	}
+
+	public void setEndDate(Date endDate) {
+		this.endDate = endDate;
+	}
+	
+	public void setStartDateString(String startDateString) {
+		this.startDateString = startDateString;
+		if (startDateString == null || startDateString.trim().equals("")) {
+			this.startDate = null;
+		} else {
+			String hiddenopenDate = JsfUtil.getStringFromParam(HIDDEN_START_DATE_FIELD);
+			Calendar tempDate = JsfUtil.convertISO8601StringToCalendar(hiddenopenDate);
+				
+			if (tempDate != null) {
+				this.startDate = tempDate.getTime();
+			}else{
+				this.startDate = null;
+			}
+
+		}
+	}
+	
+	public String getStartDateString() {
+		if (startDate == null) {
+			return "";
+		} else {
+			return JsfUtil.convertCalendarToISO8601String(startDate);
+		}
+	}
+
+	public void setEndDateString(String endDateString) {
+		this.endDateString = endDateString;
+		if (endDateString == null || endDateString.trim().equals("")) {
+			this.endDate = null;
+		} else {
+			String hiddenendDate = JsfUtil.getStringFromParam(HIDDEN_END_DATE_FIELD);
+			Calendar tempDate = JsfUtil.convertISO8601StringToCalendar(hiddenendDate);
+			if (tempDate != null) {
+				this.endDate = tempDate.getTime();
+			}else{
+				this.endDate = null;
+			}
+
+		}
+	}
+	public String getEndDateString() {
+		if (endDate == null) {
+			return "";
+		} else {
+			return JsfUtil.convertCalendarToISO8601String(endDate);
+		}
+	}
+	
```

Comment:
Sorry, I am unable to identify where spacing is different here.
Can you please help me.